### DISCUSSION TITLE
Missing caption declaration in CSS?

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#421] [Ask a question] [answered]
CSS styles lack presently a table caption declaration. It is true that (if I understand well) Markdown Extra does not provide a way to specify a caption, but for example an extension could generate a table with its caption, and a proper declaration in each style would improve the esthetical uniformity. For example for stockholm.css it could be:

```
.content caption {
    text-align: left;
    padding: 0.3em;
    border-bottom: 1px solid #ddd;
    font-weight: bold;
}
```

If it is useful, I can propose a declaration for all the themes currently supported.

**Comment by markseuffert** [comment]
Can you explain what the problem is? Please add a screenshot that shows what is missing.

**Comment by GiovanniSalmeri** [comment]
Here is a screenshot showing more or less what I mean:

![caption](https://mondodomani.org/temp/caption2.png)

Now the themes, while adding their styling for almost all HTML elements, leave \<caption\> with its default styling (i.e. simply centered), which seems a bit lame. Is this intentional? If not, perhaps it is useful (and anyway it causes no harm) to add a caption declaration to the themes. Obviously I do not mean that the styling I am proposing is the best possible.

If leaving \<caption\> with the default style is intentional, please consider this issue closed :-)

**Comment by markseuffert** [comment]
I don't know what a table caption is, to be honest. How do you make one with Markdown? 🤔

**Comment by GiovanniSalmeri** [comment]
\<table\>
\<caption\>This is a wonderful table!\</caption\>
\<tr\>..............\</tr\>
\</table\>

In Markdown Extra there is currently no way to add a caption to a table (it is possible for example in Pandoc Markdown). But obviously I can write a table in HTML and insert it in a Markdown page, or an extension (like the one I am writing :smile:) can generate a table with a caption. (An extension can add its own styling for captions, but probably this would not blend well with all the themes: whence my suggestion.)

**Comment by markseuffert** [comment]
Thank you Giovanni for the detailed explanation.

Have a look at the [highlight extension](https://github.com/annaesvensson/yellow-highlight) or the [gallery extension](https://github.com/annaesvensson/yellow-gallery). They show how to extend Markdown and how to add your own CSS. As far as I know, this is what most extensions do. Of course if table captions become part of Markdown, we can add it to the main CSS. First make it work, then make it better. :smile:

**Comment by GiovanniSalmeri** [comment, marked as answer]
Thank you very much. Yes, I am adding an CSS following the example of other extensions, but this does not solve entirely the problem: a caption styling which will appear perfect with stockholm can appear ugly with another theme. Anyway, I do understand that this is a problem which will affect one in a million user. So I consider the issue closed! :+1:
